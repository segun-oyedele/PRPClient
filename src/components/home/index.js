export * from './HomeCardContainer';
export * from './HomeCardItem';