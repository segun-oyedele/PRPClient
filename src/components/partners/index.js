export * from './SearchBar';
export * from './TableBody';