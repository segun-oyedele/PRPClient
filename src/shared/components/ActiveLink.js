import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import { useEffect, useState } from 'react';

export const ActiveLink = ({ label, href, stylesClass, navPath }) => {
  const router = useRouter();

  const [isActive, setIsActive] = useState(router.pathname === navPath);
  const activeStyles = isActive ? 'md:text-white raleway-eb cursor-default' : 'nav_link-inactive';

  const handleClick = (event) => {
    event.preventDefault();
    setIsActive(true);
    router.push(href);
  }

  useEffect(() => {
    setIsActive(router.pathname === navPath);
  }, [router.pathname]);
  

  if( isActive ) {
    return (
      <a
        className={`${ activeStyles } ${ stylesClass }`}
      >{ label }</a>
    );
  } else {
    return(
      <a
        href={ href }
        onClick={ handleClick }
        className={`${ activeStyles } ${ stylesClass }`}
      >{ label }</a>
    );
  }
};

ActiveLink.propTypes = {
  label: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  stylesClass: PropTypes.string,
  navPath: PropTypes.string.isRequired
};
