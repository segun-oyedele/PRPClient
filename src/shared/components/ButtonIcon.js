import PropTypes from 'prop-types';

const btnDefaultColors = 'bg-indigo-500 hover:bg-indigo-600 focus:bg-indigo-600 active:bg-indigo-600';
const labelDefaultColors = 'text-white';
const btnDefaultStyles = 'rounded-tr rounded-br';
const btnDefaultSpacing = 'px-4 py-2.5';

export const ButtonIcon = ({ children, onClick, btnColors, btnStyles, btnLabel, labelStyles, labelColors, buttonSpacing, iconOnRight }) => {
  return (
    <button
      className={`btn btn_icon focus:outline-none focus:ring-0 active:shadow-lg transition duration-300 ease-in-out flex items-center ${ btnStyles || btnDefaultStyles } ${ btnColors || btnDefaultColors } ${ buttonSpacing || btnDefaultSpacing }`}
      type="button"
      id="button-addon2"
      onClick={ () => onClick() }
    >
      { !iconOnRight && children }

      { btnLabel && <span className={`font-medium tracking-wide ${ labelColors || labelDefaultColors } ${ labelStyles || '' }`}>{ btnLabel }</span> }
    
      { iconOnRight && children }
    </button>
  );
};

ButtonIcon.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func.isRequired,
  btnColors: PropTypes.string,
  btnStyles: PropTypes.string,
  btnLabel: PropTypes.string,
  labelStyles: PropTypes.string,
  labelColors: PropTypes.string
};
