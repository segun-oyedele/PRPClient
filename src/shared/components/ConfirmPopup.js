import PropTypes from 'prop-types';
import { Fragment } from 'react';
import { Dialog, Transition } from '@headlessui/react';

export const ConfirmPopup = ({ isOpen, setIsOpen, onClick, title, description, colorsPrimaryBoton, buttonTitle, showButton = true }) => {

  const colorsBtn = colorsPrimaryBoton || "bg-indigo-500 border border-transparent rounded-md hover:bg-indigo-600 focus:bg-indigo-600 active:bg-indigo-600 text-white";

  function closeModal() {
    setIsOpen(false);
  }

  const handleClick = () => {
    closeModal();
    onClick();
  }

  return (
    <>
      <Transition appear show={isOpen} as={Fragment}>
        <Dialog as="div" className="relative z-50" onClose={closeModal}>
          <Transition.Child
            as={Fragment}
            enter="ease-out duration-300"
            enterFrom="opacity-0"
            enterTo="opacity-100"
            leave="ease-in duration-200"
            leaveFrom="opacity-100"
            leaveTo="opacity-0"
          >
            <div className="fixed inset-0 bg-black bg-opacity-25" />
          </Transition.Child>

          <div className="fixed inset-0 overflow-y-auto">
            <div className="flex items-center justify-center min-h-full p-4 text-center">
              <Transition.Child
                as={Fragment}
                enter="ease-out duration-300"
                enterFrom="opacity-0 scale-95"
                enterTo="opacity-100 scale-100"
                leave="ease-in duration-200"
                leaveFrom="opacity-100 scale-100"
                leaveTo="opacity-0 scale-95"
              >
                <Dialog.Panel className="w-full max-w-md p-6 overflow-hidden text-left align-middle transition-all transform bg-white shadow-xl rounded-2xl">
                  <Dialog.Title
                    as="h3"
                    className="text-lg font-medium leading-6 text-gray-900"
                  >
                    { title }
                  </Dialog.Title>
                  <div className="mt-2">
                    <p className="text-sm text-gray-500">
                      { description }
                    </p>
                  </div>
                  { showButton &&
                    <div className="grid grid-cols-2 gap-2 mt-4">
                      <button
                        type="button"
                        className={`${ colorsBtn } inline-flex justify-center px-4 py-2 text-sm font-medium transition duration-300 border border-transparent rounded-md focus:outline-none focus-visible:ring-2`}
                        onClick={ handleClick }
                      >
                        { buttonTitle }
                      </button>
                      <button
                        type="button"
                        className="inline-flex justify-center px-4 py-2 text-sm font-medium text-indigo-600 transition duration-300 bg-white border border-transparent rounded-md hover:border-indigo-500 focus:border-indigo-500 active:border-indigo-500 focus:outline-none focus-visible:ring-2 "
                        onClick={ closeModal }
                      >
                        Cancel
                      </button>
                    </div>
                  }
                </Dialog.Panel>
              </Transition.Child>
            </div>
          </div>
        </Dialog>
      </Transition>
    </>
  );
};

ConfirmPopup.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  setIsOpen: PropTypes.func.isRequired,
  onClick: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  colorsPrimaryBoton: PropTypes.string,
  buttonTitle: PropTypes.string.isRequired,
  showButton: PropTypes.bool
}