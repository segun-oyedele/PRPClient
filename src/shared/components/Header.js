import { Navbar } from './';
import Link from 'next/link';
import Image from 'next/image';
import { useRouter } from 'next/router';
import { navLinks } from '../data';
import { useEffect, useState } from 'react';

export const Header = () => {
  
  const router = useRouter();
  const [isActive, setIsActive] = useState(router.pathname === navLinks[0].navPath);

  useEffect(() => {
    setIsActive(router.pathname === navLinks[0].navPath);
  }, [router.pathname]);

  return (
    <header className="sticky top-0 z-10 md:top-auto md:static md:z-auto header-tag primary-blue">
      <div className="container flex items-center h-full min-w-full p-5 md:p-0 md:px-4">
        { isActive ?
          <a className="flex items-center gap-3 cursor-default">
            <Image
              alt='Last Mile Logo'
              className="w-16"
              width={106}
              height={50}
              src={`${process.env.iisPath}/img/last_mile_logo.png`}
              title="Last-Mile-Logo"
            />
          </a>
          :
          <Link href={`${navLinks[0].to}`}>
            <a className="flex items-center gap-3">
              <Image
                alt='Last Mile Logo'
                className="w-16"
                width={106}
                height={50}
                src={`${process.env.iisPath}/img/last_mile_logo.png`}
                title="Last-Mile-Logo"
              />
            </a>
          </Link>
        }

        <Navbar />
      </div>
    </header>
  );
};
