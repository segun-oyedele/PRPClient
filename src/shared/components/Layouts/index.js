import Head from 'next/head';
import PropTypes from 'prop-types';
import { Header } from '../';
import { Sidebar } from '../';

export const Layout = ({ children, headTitle }) => {
  return (
    <>
      <Sidebar />
      <Head>
        <title>{`Partner Reporting Package - ${ headTitle }`}</title>
        <meta name="description" content="CDL Last Mile partner reporting package" />
        <link rel="apple-touch-icon" sizes="152x152" href={`${process.env.iisPath}/apple-touch-icon.png`} />
        <link rel="icon" type="image/png" sizes="32x32" href={`${process.env.iisPath}/favicon-32x32.png`} />
        <link rel="icon" type="image/png" sizes="16x16" href={`${process.env.iisPath}/favicon-16x16.png`} />
        <link rel="manifest" href={`${process.env.iisPath}/site.webmanifest`} />
        <link rel="mask-icon" href={`${process.env.iisPath}/safari-pinned-tab.svg`} color="#5bbad5" />
        <meta name="msapplication-TileColor" content="#da532c" />
        <meta name="theme-color" content="#ffffff" />
      </Head>

      <Header />

      <div className="container px-4">
        {children}
      </div>
    </>
  );
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  headTitle: PropTypes.string.isRequired
};
