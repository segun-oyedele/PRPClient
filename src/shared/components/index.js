export * from './ActiveLink';
export * from './BurgerIcon';
export * from './ButtonIcon';
export * from './Header';
export * from './Layouts';
export * from './Links';
export * from './Navbar';
export * from './PageTitle';
export * from './Pagination';
export * from './Sidebar';
export * from './Table';
export * from './TableHead';
