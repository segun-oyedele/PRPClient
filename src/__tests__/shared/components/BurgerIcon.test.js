import { render, screen, fireEvent } from '@testing-library/react';
import { BurgerIcon } from '@/shared/components/BurgerIcon';

describe('Test on BurgerIcon component', () => {
  
  test('should match with snapshot', () => {
    const { container } = render(<BurgerIcon handleIsOpen={ jest.fn() } isOpen={ false } genericHamburgerLine="" />);
    expect( container ).toMatchSnapshot();
  });

  test('should to call handleIsOpen', () => {
    const handleIsOpen = jest.fn();

    render(<BurgerIcon handleIsOpen={ handleIsOpen } isOpen={ false } /> );
    const button = screen.getByRole('button');
    fireEvent.click(button);

    
    expect( handleIsOpen ).toHaveBeenCalled();
    expect( handleIsOpen ).toHaveBeenCalledTimes(1);
  });
  
})