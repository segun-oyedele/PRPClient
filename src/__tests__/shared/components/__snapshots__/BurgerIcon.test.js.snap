// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on BurgerIcon component should match with snapshot 1`] = `
<div>
  <button
    class="fixed z-50 flex flex-col items-center justify-center w-12 h-12 bg-white shadow-md border-2 border-black transition-opacity duration-300 rounded group right-5 top-5 md:hidden hover:opacity-100 sidebar opacity-50 "
  >
    <div
      class=" group-hover:opacity-100 sidebar"
    />
    <div
      class=" group-hover:opacity-100 sidebar"
    />
    <div
      class=" group-hover:opacity-100 sidebar"
    />
  </button>
</div>
`;
